core/src/main/java/org/wso2/carbon/kernel/configresolver/ConfigResolverUtils.java

```diff
@@ -212,9 +210,10 @@ public static String convertXMLtoString(Document doc) {
      * @return XML formatted String
      */
     public static String convertXMLSourceToString(Source source) {
+        StringWriter stringWriter = null;
         String xmlString;
         try {
-            StringWriter stringWriter = new StringWriter();
+            stringWriter = new StringWriter();
```

Comment:
We'll have to use nested try-catch blocks if we want to use auto closeable. But it is better to have a single try-catch block here and close the StringWriter in the finally block.